### DISCUSSION TITLE
Passing props to components in testing

### DISCUSSION THREAD

**Discussion by takdw** [livewire/livewire#1863] [1. Help]
Not sure if I'm missing something but I'm unable to test a component that accepts props from Laravel. Let's use a simple example to demonstrate what I'm having issue with. 
```php
class PostManagement extends Component
{
    public $category;

    public $postBeingUpdated;

    public function render()
    {
        return view('livewire.post-management', [
            'cards' => $this->category->posts()->paginate(10),
        ]);
    }

    public function updateStatus()
    {
        $validated = $this->validate([
            'status' => 'required'
        ]);

        $post = Post::find($this->postBeingUpdated);
        $post->status = $this->status;
        $post->save();
    }
}
```
Somewhere, in a Laravel Blade file, I use this component as `<livewire:post-management :category="$category" />`.

Now I'm trying to test the `updateStatus` functionality.
```php
class UpdatingPostStatusTest extends TestCase
{
    use RefreshDatabase;

    /** @test */
    public function canUpdateTheStatusOfACard()
    {
        $category = Category::factory()->create();
        $post = Post::factory()->create(['category_id' => $category->id]);

        Livewire::test(PostManagement::class)
            ->set('category', $category)
            ->set('status', Post::REJECTED)
            ->call('updateStatus');

        $this->assertEquals(Post::REJECTED, $post->fresh()->status);
    }
}
```
When I try to run this test I get the error `Call to a member function posts() on null`. `$category` isn't being set and is `null` when I try to paginate its `posts`. Is there a reason why this is happening? Am I doing something wrong? I just started out Livewire yesterday and I've sorta hit a road block here (when it comes to testing at least).

**Comment by joshhanley** [comment]
You can pass props to the Livewire::test method. See example from the docs under all available test methods section https://laravel-livewire.com/docs/2.x/testing

```php
Livewire::test('foo', ['bar' => $bar]);
// Test the "foo" component with "bar" set as a parameter.
```

Hope this helps!
